extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/MSQCompactionRunner.java

```diff
@@ -143,6 +143,12 @@ public CompactionConfigValidationResult validateCompactionTask(
       Map<Interval, DataSchema> intervalToDataSchemaMap
   )
   {
+    if (intervalToDataSchemaMap.size() > 1) {
+      return CompactionConfigValidationResult.failure(
+          "MSQ: Disjoint compaction intervals[%s] not supported",
```

Comment:
Thanks for adding this! Please add a comment on why this is invalid.

Reply:
Added.